src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -183,17 +183,31 @@ private int TryGetAppSettingAsIntWithDefault(string key, int defaultValue)
 
         public object AgentRunId { get { return _serverConfiguration.AgentRunId; } }
 
+        private bool _agentEnabled;
+
+        private static string _agentEnabledAppSettingString;
+        private static bool? _agentEnabledAppSettingRead;
+        private static bool _agentEnabledAppSettingParsed;
+
+
         public virtual bool AgentEnabled
         {
             get
             {
-                var agentEnabledAsString = _configurationManagerStatic.GetAppSetting("NewRelic.AgentEnabled");
+                // read from app setting one time only and cache the result in a static
+                if (!_agentEnabledAppSettingRead.HasValue)
```

Comment:
I think this can probably be simplified a bit? I don't think `_agentEnabledAppSettingRead` needs to be nullable, and `_agentEnabledAppSettingString` can be a local var.

Reply:
I cleaned things up a bit.